File: spec/RoundSpec.js
describe('frame', function() {
  beforeEach(function() {
    frame = new Frame();
  });

  it('stores pins boweld', function() {
    frame.bowl(8)
    frame.bowl(1)
    expect(frame.score).toEqual(9)
  });

  it('knows if it\'s a strike', function() {
    frame.bowl(10)
    expect(frame.strike).toBe(true)
    expect(frame.spare).toBe(false)
  });

  it('knows if it\'s a spare', function() {
    frame.bowl(5)
    frame.bowl(5)
    expect(frame.spare).toBe(true)
    expect(frame.strike).toBe(false)
  });

  it('knows the frame is over', function() {
    frame.bowl(10)
    expect(frame.end).toBe(true)
  });

  it('cannot have a score higher than 10', function() {
    frame.bowl(5)
    frame.bowl(6)
    expect(frame.score).toEqual(5)
  });

});

File: spec/round_spec.rb
describe Frame do

  let(:not_a_strike) { Frame.new(5, 4)}
  let(:strike) { Frame.new(10, 0) }
  let(:spare) { Frame.new(7, 3) }

  it 'stores bowls in a hash' do
  expect(not_a_strike.frame[:first_throw]).to eq 5
  expect(not_a_strike.frame[:second_throw]).to eq 4
  end

  it '#is_strike? method changes strike status from false to true' do
    expect(strike.strike).to eq false
    strike.is_a_strike?
    expect(strike.strike).to eq true
  end

  it '#is_strike? method does not change strike status unless it is a strike' do
    expect(not_a_strike.strike).to eq false
    not_a_strike.is_a_strike?
    expect(not_a_strike.strike).to eq false
  end

  it '#is_spare? method changes spare status from false to true' do
    expect(spare.spare).to eq false
    spare.is_a_spare?
    expect(spare.spare).to eq true
  end

  it '#is_spare? method does not change spare status unless it is a spare' do
    expect(strike.spare).to eq false
    strike.is_a_spare?
    expect(strike.spare).to eq false
  end
end